include/hal/library/memlib.h

```diff
@@ -28,9 +28,9 @@
  *   5) "len" or "dst_len" is greater than (MAX_ADDRESS - "src_buf" + 1).
  *   6) "len" or "dst_len" is greater than (SIZE_MAX >> 1).
  *
- * In case of error, the "dst_buf" is left unmodifed.
- * This behavior is different than C11 memcopy_s, which zeros the "dst_buf"
- * if there is a runtime violation.
+ * In case of error, "dst_len" bytes of "dst_buf" is zeroed, if "dst_buf"
+ * points to a non-NULL value and "dst_len" does not exceed
+ * ((MAX_ADDRESS - "dst_buf" + 1) or (MAX_ADDRESS - "src_buf" + 1) or (SIZE_MAX >> 1)).
  *
```

Comment:
I suggest you can mention, it is aligned to C11 standard

Please double check MSFT memcpy_s() - https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/memcpy-s-wmemcpy-s, to see they are aligned.
Also also mention that in the comment, in case someone wants to use C11 version or MSFT version memcpy_s().

Reply:
If memcopy_s is different in windows and linux world, how we can implement copy_mem_s() via memcopy_s() ?

Reply:
I was going to follow the cppreference [description](https://en.cppreference.com/w/c/string/byte/memcpy) of memcpy_s and document the differences with MS and safeclib. In particular, cppreference does not allow for NULL dest or source, and only says non-zero is returned on error, rather than a specific error code. 

If clients want to call safeclib or the microsoft version, they need to be aware of the differences. I will push a change you can review.